pipeline-runner/R/gem2s-2-load_user_files.R

```diff
@@ -239,6 +209,40 @@ parse_rhapsody_matrix <- function(config, input_dir) {
   return(list(counts_list = counts_list, annot = annot))
 }
 
+read_10x_annotations <- function(annot_fpath, sample) {
+  gene_column <- 1
+
+  annot <- read.delim(annot_fpath, header = FALSE)
+
+  # Some feature files have less columns than expected.
+  # Duplicate first column if there is only one column with gene names/ids, or
+  # if there is a "Gene Expression" column
+  if (ncol(annot) == 1 || annot[1, 2] == "Gene Expression") {
+    annot[, 2] <- annot[, 1]
+  }
+
+  feature_types <- get_feature_types(annot)
+
+  message("Feature types are ", feature_types, "for sample ", sample)
+
+  # reverse annot cols if symbols are first
+  if (feature_types == SYM_IDS) {
+    annot[, c(1, 2)] <- annot[, c(2, 1)]
+    # Read10x reads from the feature file which is not modified, but contains IDS in the second column. Use gene_column to indicate to Read10x from where to read the ids. 
+    gene_column <- 2
+    feature_types <- IDS_SYM
+  }
+
+  # make same names in annot seurat makes for count matrix.
+  # Only first col, we either copy or reverse order of the annot data.frame
```

Comment:
```suggestion
  # Make the names in annot the same as the ones in the Read10x generated count matrix
  # Since Seurat uses makes.unique, we need to as well.
  # Only for the first column, at this stage column 1 should be IDs.
```